tests/test_url_config.py

```diff
@@ -60,5 +71,24 @@ def test_chained(self):
         p = self.st.get_single_plugin(nt_test_run_id, 'test_data')
         self.assertEqual(p.test_config, 1)
 
+    def test_cache(self):
+        p = self.st.get_single_plugin(nt_test_run_id, 'test_data')
+        self.assertEqual(p.cached_config, 666)
+        self.st.set_config({'cached_config': 'random://abc'})
+        p = self.st.get_single_plugin(nt_test_run_id, 'test_data')
+        cached_value = p.cached_config
+        self.assertEqual(cached_value, p.cached_config)
+
+        # test if previous value is evicted, since cache size is 1
+        self.st.set_config({'cached_config': 'random://dfg'})
+        p = self.st.get_single_plugin(nt_test_run_id, 'test_data')
+        self.assertEqual(p.cached_config, p.cached_config)

```

Comment:
can this ever be not true?

Reply:
it can be if the caching isnt working ;) I'm doing a sneaky trick using a protocol that returns a random number so if the value is evaluated twice it will get a different number but if it returns the cached value it will be the same :) I guess i could change it to a counter if you are afraid of the random function returning the same value twice in a row...

Reply:
ah okay, I see, thanks, mind adding this as a comment, just to be extra clear?

Reply:
done.